### PR TITLE
fix(velocity): prevent velocity 1.7 from coming through and use 2.3

### PR DESCRIPTION
**Pull request by RyanHolstien**
Upgrades Velocity to remove vulnerabilities CVE-2020-13936 & CVE-2020-13959

### CODE REVIEW

File: metadata-io/build.gradle [hunk 1]
```diff
@@ -18,7 +18,8 @@ dependencies {
   compile spec.product.pegasus.generator
 
   compile externalDependency.dgraph4j exclude group: 'com.google.guava', module: 'guava'
-  compile externalDependency.lombok
+  compileOnly externalDependency.lombok
+  implementation externalDependency.commonsCollections
```
Comment (jjoyce0510) [inline metadata-io/build.gradle:22]: is this serving as replacement for velocity?

File: metadata-events/mxe-avro-1.7/build.gradle [hunk 2]
```diff
@@ -7,7 +7,9 @@ apply plugin: 'java'
 
 dependencies {
   compile externalDependency.avro_1_7
-  compile externalDependency.avroCompiler_1_7
+  compile(externalDependency.avroCompiler_1_7) {
+    exclude group: 'org.apache.velocity', module: 'velocity'
```
Comment (jjoyce0510) [inline metadata-events/mxe-avro-1.7/build.gradle:11]: do we know of any adverse effects of excluding this altogether? causes potential for no class found exceptions at runtime

Comment (jjoyce0510) [reply metadata-events/mxe-avro-1.7/build.gradle:11]: (but maybe avroCompile only used at build time?)

File: metadata-io/build.gradle (hunk 1 above)
Comment (RyanHolstien) [reply metadata-io/build.gradle:22]: I think maybe we were utilizing a transitive dependency from velocity on Apache Commons Collections. This just covers the previous build failure where CollectionUtils was not found.

File: metadata-events/mxe-avro-1.7/build.gradle (hunk 2 above)
Comment (RyanHolstien) [reply metadata-events/mxe-avro-1.7/build.gradle:11]: So at runtime we're including the updated version, in the WAR runtimeClasspath Velocity-Core 2.3 is present (per dependencyInsight). This excludes the legacy Velocity dep and would impact build time only (barring 2.3 actually being incompatible in which case we cannot upgrade it in place and would need to update to Avro 1.9+)

**Review by shirshanka — APPROVED** [review]
LGTM

**Pull request merged**